python_modules/dagster/dagster_tests/core_tests/execution_tests/test_asset_execution_context.py

```diff
@@ -0,0 +1,136 @@
+import inspect
+import warnings
+
+import pytest
+from dagster import AssetExecutionContext, OpExecutionContext, asset, materialize
+from dagster._core.execution.context.compute import _get_deprecation_kwargs
+
+
+def test_deprecation_warnings():
+    # Test that every method on OpExecutionContext is either reimplemented by AssetExecutionContext
+    # or throws a deprecation warning on AssetExecutionContext
+
+    # If this test fails, it is likely because you added a method to OpExecutionContext without
+    # also adding that method to AssetExecutionContext. Please add the same method to AssetExecutionContext
+    # with an appropriate deprecation warning (see existing deprecated methods for examples).
+    # If the method should not be deprecated for AssetExecutionContext, please still add the same method
+    # to AssetExecutionContext and add the method name to asset_execution_context_attrs
+
+    asset_execution_context_attrs = [
+        "op_execution_context",
+        "is_asset_step",
+        "asset_key",
+        "asset_keys",
+        "provenance",
+        "provenance_by_asset_key",
+        "code_version",
+        "code_version_by_asset_key",
+        "is_partition_step",
+        "partition_key",
+        "partition_key_range",
+        "partition_time_window",
+        "run_id",
+        "job_name",
+        "retry_number",
+        "dagster_run",
+        "pdb",
+        "log",
+        "log_event",
+        "assets_def",
+        "selected_asset_keys",
+        "get_asset_provenance",
+        "get_assets_code_version",
+        "asset_check_spec",
+        "partition_key_range_for_asset_key",
+        "instance",
+        # TODO - potentially remove below if they get deprecated
+        "resources",
+        "run_config",
+    ]
+
+    other_ignores = [
+        "_abc_impl",
+        # TODO - what to do about instance vars for OpExecutionContext? listed below
```

Comment:
The below attributes are instance variables on `OpExecutionContext` that are not available on `AssetExecutionContext` because `AssetExecutionContext` overrides the `__init__` of `OpExecutionContext`. We could:
* make properties for each of these attrs on `AssetExecutionContext` and throw deprecation warnings
* do nothing since the underscore implies they are private. But there's probably a user somewhere who is calling these, and their code would break

Reply:
IMO we should be able to break users that are accessing private members